test/test_core/test_tree_policies/test_UCT_TreePolicy.cpp

```diff
@@ -0,0 +1,48 @@
+#include <UCT_TreePolicy.h>
+#include <gtest/gtest.h>
+#include <string>
+
+class UCT_TreePolicyTest : public ::testing::Test {
+  protected:
+    void SetUp() override {
+        treePolicy = UCT_TreePolicy(simpleExpand, simpleChild);
+
+        // root
+        State<string> rootNode = State<string>("rootNode");
+        root = SearchNode(NULL, rootNode, false);
+
+        // root as parent
+        State<string> rootChild1 = State<string>("rootChild1");
+        State<string> rootChild2 = State<string>("rootChild2");
+        SearchNode rootChild1 = SearchNode(root, rootChild1, false);
+        SearchNode rootChild2 = SearchNode(root, rootChild2, false);
+    }
+
+    SearchNode root;
+    UCT_TreePolicy treePolicy;
+
+    SearchNode simpleExpand(SearchNode &node){
+        State<string> state = node.unvisited_child_states.at(0);
+        SearchNode expanded_node = SearchNode(node, state, false);
+        return expanded_node;
+    }
+
+    SearchNode simpleChild(SearchNode &node){
```

Comment:
this function can be inlined to avoid a function call in the compiled code
```suggestion
    inline SearchNode simpleChild(SearchNode &node){
```